src/Pact/Types/ChainMeta.hs

```diff
@@ -78,14 +89,16 @@ instance Serialize PrivateMeta
 
 -- | Contains all necessary metadata for a Chainweb-style public chain.
 data PublicMeta = PublicMeta
-  { _pmChainId :: ChainId
-  , _pmSender :: Text
-  , _pmGasLimit :: GasLimit
-  , _pmGasPrice :: GasPrice
+  { _pmChainId :: !ChainId
+  , _pmSender :: !Text
+  , _pmGasLimit :: !GasLimit
+  , _pmGasPrice :: !GasPrice
+  , _pmTtl :: !TTLSeconds
```

Comment:
I think if you're going to call the type `TTLSeconds` you might as well call this `_pmTTL`

Reply:
I considered this, but the JSON instance for `PublicMeta` then has to be written by hand, since the `lensyToJSON`-based instance produces `TTL: 0`. From a UX perspective that's pretty bad - i figured `ttl: 0` was much more amenable, and we engineers could bite the bullet. Unless you wanted to write the instance by hand, that is.

Reply:
@slpopejoy unfortunately the way `lensyToJSON` acts on names is by camelcase, so it parses `lensyToJSON 3 _pmTTL` as a JSON field called "tTL". 

```haskell
λ> lensyToJSON 3 (def :: PublicMeta)
Object (fromList [("creationTime",Number 0.0),("tTL",Number 0.0),("gasLimit",Number 0.0),("chainId",String ""),("gasPrice",Number 0.0),("sender",String "")])
```

Which is v. bad UI/UX imo. If we want to support `_pmTTL` and have it parse the field "ttl", we have to write the instance by hand, which I was hoping to avoid by just biting the dev bullet on the name. If you're okay with rewriting the `PublicMeta` To/FromJSON instance by hand, I'll go ahead and do it.

Reply:
Added the handwritten parser